rfcs/0000-ckb-softfork-activation/0000-ckb-softfork-activation.md

```diff
@@ -0,0 +1,132 @@
+---
+Number: "0000"
+Category: Standards Track
+Status: Draft
+Author: Dingwei Zhang <zhangsoledad@gmail.com>
+Created: 2022-06-16
+---
+
+# CKB softfork activation
+
+## Abstract
+
+This document specifies a proposed change to the semantics of the 'version' field in CKB blocks, allowing multiple backward-compatible changes (further called "softforks") to be deployed in parallel. It relies on interpreting the version field as a bit vector, where each bit can be used to track an independent change. These are tallied in each epoch. Once the consensus change succeeds or times out, there is a "fallow" pause, after which the bit can be reused for later changes.
+
+## Specification
+
+### Parameters
+
+Each softfork deployment is specified by the following per-chain parameters (further elaborated below):
+
+1. The `name` specifies a very brief description of the softfork, reasonable for use as an identifier.
+2. The `bit` determines which bit in the `version` field of the block is to be used to signal the softfork lock-in and activation. It is chosen from the set {0,1,2,...,28}.
+3. The `start_epoch` specifies the first epoch in which the bit gains meaning.
+4. The `timeout_epoch` specifies an epoch at which the miner signaling ends. Once this epoch has been reached, if the softfork has not yet locked_in (excluding this epoch block's bit state), the deployment is considered failed on all descendants of the block.
+5. The `threshold` specifies the minimum ratio of block per epoch, which indicate the locked_in of the softfork during the epoch.
+6. The `minimum_activation_epoch` specifies the epoch at which the softfork is allowed to become active.
```

Comment:
I suggest rename `minimum_activation_epoch` to `activation_epoch`. With a `minimum_` prefix, it implies "the softfork can be activated at any epoch after this epoch". But the wanted behavior is to activate the softfork immediately at this activation epoch.

Reply:
Just found that `minimum_activation_epoch` is actually accurate after reading BIP8 and BIP9 again.

Reply:
There is a similar discussion here, https://github.com/bitcoin/bitcoin/pull/21377/files#r598896420